### PR TITLE
Complete

### PR DESCRIPTION
**Pull request by levialmog**
Ready for your review

### CODE REVIEW

File: piece_of_cake.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+def get_recipe_price(prices=None, optionals=None, **ingredients):
```
Comment (talbarfm) [inline piece_of_cake.py]: 'why do you set prices default to be None? can it actually be None?

File: piece_of_cake.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        if optionals is None:
+            optionals = []
+
+        shopping_cart = 0
+        for price in prices:
+            if price not in optionals:
+                num_for_mul = ingredients[price]/100
+                shopping_cart = shopping_cart + int(prices[price])*num_for_mul
+        print(int(shopping_cart))
+    else:
```
Comment (talbarfm) [inline piece_of_cake.py]: You dont have to validate input - if you request parameters and the user doesnt supplies them, or supplies other types, its their problem, and the code will fail  - you have to handle all the legit values - an empty dictionary of ingredients is a legit value - None is not  a legit value

Comment (talbarfm) [reply piece_of_cake.py]: Another legit option - what happens if an ingredient is not in prices

File: piece_of_cake.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+        print("Bad input")
+
+
+get_recipe_price({'chocolate': 18, 'milk': 8}, chocolate=200, milk=100)
```
Comment (talbarfm) [inline piece_of_cake.py]: https://www.freecodecamp.org/news/if-name-main-python-example/ 
Read this and add it - dont just call functions outside of any scope

File: piece_of_cake.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+def get_recipe_price(prices=None, optionals=None, **ingredients):
+
+    if prices is not None and ingredients is not None and len(ingredients) == len(prices):
```
Comment (talbarfm) [inline piece_of_cake.py]: You dont have to check if prices is not None - just assume the user is not stupid - check if its not empty though and then return a matching message - or raise a matching exception

File: piece_of_cake.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+        if optionals is None:
+            optionals = []
+
+        shopping_cart = 0
+        for price in prices:
+            if price not in optionals:
+                num_for_mul = ingredients[price]/100
```
Comment (talbarfm) [inline piece_of_cake.py]: spaces around operators

Comment (talbarfm) [reply piece_of_cake.py]: apply everywhere

**Review by talbarfm — CHANGES_REQUESTED** [review]

**Review by lovodkin93 — APPROVED** [review]
looks much better.

**Pull request merged**